diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -940,26 +946,46 @@ int DifferentialFunctionComparator::cmpBasicBlocks(
             // it will be used in an inverse condition. Hence, we skip it here
             // and mark that it may be inverse-matching the condition that
             // has been originally mapped to the operand of the not operation.
-            if (isLogicalNot(&*InstL) || isLogicalNot(&*InstR)) {
+            if (config.Patterns.InverseConditions
+                && (isLogicalNot(&*InstL) || isLogicalNot(&*InstR))) {
+                // Note: This does not handle cases with a lot of `not`:
+                // %1 = cmp true, false | %1 = cmp true, false
+                //                      | %2 = not %1 (%2 = xor %1, true)
+                //                      | %3 = not %2 (%3 = xor %2, true)
+                //                      | %4 = not %3 (%4 = xor %3, true)
+                // br %1, l %T, l %F    | br %4, l %F, l %T
                 sn_mapL.erase(&*InstL);
                 sn_mapR.erase(&*InstR);
```

Comment:
Also I am not sure, if this should not erase only the `not` instruction or maybe even the original `cmp` instruction?

Reply:
Thanks for the explanation, it makes sense. Now I am thinking if it should not also be erased from `mappedValuesBySn`, but I think it is not necessary, because the next instructions will overwrite it .

Reply:
That's correct but I think that we should erase it anyways - just to be sure that there is nothing left from the (incorrect) comparison of the last 2 instructions. Ideally, we'd use `undoLastInstCompare` but we can't because we don't want to control the `InstL` and `InstR` iterators in a slightly different way.

Good catch btw! Can you please add the `mappedValuesBySn` erase?

Reply:
Thanks, good idea.